### PR TITLE
Bucket Policy: Fixes and tests. 

### PR DESCRIPTION
**Pull request by hackintoshrao**
With changes in code sending this as a separate commit. 
- Addition to allowed actions.
- Changes to ResourceMatch logic.
- Test for action match function.

### CODE REVIEW

File: bucket-policy-parser.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ const (
 
 // supportedActionMap - lists all the actions supported by minio.
 var supportedActionMap = map[string]struct{}{
+	"*":                             {},
+	"s3*":                           {},
```
Comment (harshavardhana) [inline bucket-policy-parser.go]: This should be "s3:*" , can you check?

**Comment by harshavardhana** [comment]
Can you also paste new coverage following these changes? - will see about automating this for minio server as well.

**Comment by hackintoshrao** [comment]
> Can you also paste new coverage following these changes?

The coverage is still bad since the GET,PUT and DELETE bucket handlers needs tests. Working on it. 

> will see about automating this for minio server as well.

What do you mean by this ? I'll be adding some functional tests as well with these changes.

File: bucket-policy-parser.go (hunk 1 above)
Comment (hackintoshrao) [inline bucket-policy-parser.go]: Changing it to `s3:*`

**Comment by harshavardhana** [comment]
> What do you mean by this ? I'll be adding some functional tests as well with these changes.I

Will add support for running coverage all the time, so that developers get to see if their patch decreases the coverage such patches will not be accepted.  (there are some exceptions though)

**Comment by hackintoshrao** [comment]
@harshavardhana : As you pointed out there are exceptions.

**Comment by hackintoshrao** [comment]
Part of fix for https://github.com/minio/minio/issues/1990

**Pull request merged**